App.js

```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
 
 const getData = async () => {
 	  try {
-	const jsonValue = await AsyncStorage.getItem('@tasks')
-	 return jsonValue != null ? JSON.parse(jsonValue) : null;
-        } catch(err) {
-		console.log(err);
+      const jsonValue = await AsyncStorage.getItem('@tasks')
+      
+      console.log('inside get promise', jsonValue, JSON.parse(jsonValue))
+      return jsonValue != null ? JSON.parse(jsonValue) : null;
+    } catch(err) {
+      console.log(err);
 	  }
-	}
-			    
+}
+  
+ const addTodo = () => {
+  let id = todos[todos.length - 1]?.id || 0 ;
+   let thisId = id * 1 + 1 + "";
+   const updatedToDoArray = [...todos, { id: thisId, text }];
+  setTodos(updatedToDoArray);
+  storeData(updatedToDoArray)
+  onChangeText("");
+ }
+ 
 
-const remove = id => { 
-setTodos(() =>todos.filter(item => item.id !== id )) ; 
+  const remove = id => { 
+    const updatedToDoArray = todos.filter(item => item.id !== id);
+    setTodos(updatedToDoArray);
```

Comment:
You wrote the filter function INSIDE the useState hook. Your useState hook is set to hang on to an Array. You can pass an Array to it, you really shouldn't pass a function.